### PR TITLE
chore: Ensure binaries are downloadable after release

### LINKED ISSUE #16689: Add command to release process to ensure release binary is downloadable
Add a command for the release process that tests the binary url for the newly released version of Cypress that ensures it is available and downloadable on each operating system.

### PR DESCRIPTION
**Pull request by chrisbreiding**
- Closes #16689

### User facing changelog

N/A - internal only

### Additional details

- Checks the URLs for each of the Cypress binary downloads as part of the `binary-release` step of the release process. Will fail if the binary URL 404s. Failure looks like this:

	Screen Shot 2021-05-28 at 4 13 10 PM

- Adds `yarn binary-ensure` so it can be manually run as well.

- Purges the cloudflare cache as part of the `move-binaries` step of the release process.

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Such a beautiful error message with clear next steps 😍

**Comment by jennifer-shehane** [comment]
@chrisbreiding The unit tests are failing. Need to make sure the tests run properly for outside contributors if it requires credentials. 

Screen Shot 2021-06-01 at 8 30 30 AM

File: scripts/binary/index.js [hunk 1]
```diff
@@ -143,12 +144,82 @@ const deploy = {
...
+      { platform: 'win32', arch: 'x64' },
+    ]
+
+    const checkSystem = ({ platform, arch }) => {
+      const url = `https://download.cypress.io/desktop/${version}?platform=${platform}&arch=${arch}`
+      const system = `${platform}-${arch}`
+
+      process.stdout.write(`Checking for ${chalk.yellow(system)} at ${chalk.cyan(url)} ... `)
+
+      return urlExist(url)
```
Comment (flotwig) [inline scripts/binary/index.js:170]: mild nit, why not use request-promise here? it will throw on a non-200 error code

i have a dep vendetta

**Review by flotwig — APPROVED** [review]
very nice

File: scripts/binary/index.js (hunk 1 above)
Comment (chrisbreiding) [reply scripts/binary/index.js:170]: Good point. All it seems to do is the equivalent of `request.head()`, so I can change this to do the same without the extra dep.

**Pull request merged**